src/main/java/UI/ListingListPage.java

```diff
@@ -1,129 +1,100 @@
+            //info needed to create a ListingPanel for each listing
+            JButton listingDetails = new JButton(listing.getTitle());
+            JLabel resultPrice = new JLabel("Price: " + listing.getPrice());
+            JLabel resultImage = new JLabel(listing.getImagePath());
+            JLabel resultDescription = new JLabel(listing.getDescription());
+            JLabel resultDate = new JLabel(listing.getDate().toString());
+
+            //creating individual ListingPanel element
+            ListingPanel listingInfo = new ListingPanel(listingDetails, resultPrice, resultImage, resultDescription,
+                    resultDate);
+
+            this.add(listingInfo);
+
+            //align listing in the middle
+            LAYOUT.putConstraint(SpringLayout.WEST, listingInfo, 0, SpringLayout.WEST, SearchBar);
+
+            //align listing vertically
+            if (previousListing == null) {
+                //is the first listing --> put under titleLabel
+                LAYOUT.putConstraint(SpringLayout.NORTH, listingInfo, 30, SpringLayout.SOUTH, titleLabel);
+                previousListing = listingInfo; //set previousListing to the one we just created
+            }
+            else {
+                //not the first listing --> put under previousListing
+                LAYOUT.putConstraint(SpringLayout.NORTH, listingInfo, 30, SpringLayout.SOUTH, previousListing);
+                previousListing = listingInfo; //reset previousListing to the one we just created
+            }
         }
-
-
-        setVisible(true);
     }
-//    public void paint(Graphics g) {
-//        for (int i = 0; i < 10; i++) {
-//            g.drawRect(x_length, y_height + i * 100 - 200, box_length, box_height);
-//        }
-//    }
 
-
-    // method actionPerformed()
-    // to get the action performed
-    // by the user and act accordingly
+    @Override
     public void actionPerformed(ActionEvent e) {
-//        if (e.getSource() == logo) {
-//              reroute to listings page
-//        } else if (e.getSource() == cart) {
-//              reroute to cart
-//        }
-//        else if (e.getSource() == checkout) {}
-        // reroute to checkout
-//        else if (e.getSource() == profile) {}
-        // reroute to profile
-//        else if (e.getSource() == listing) {}
-        // reroute to listing detail
-
-    }
-    public List<Listing> showListings() throws IOException {
-        DatabaseController controller = new DatabaseController();
-        List <Listing> listings = controller.getAllListings();
-        return listings;
-    }
-
-    public static void main(String[] args) {
-        new ListingListPage();
-
-
+        if (e.getSource() == SEARCH) {
+            SearchForm form = new SearchForm(jtSearch.getText(), controller);
+            SearchResponseModel responseModel = form.getResponseModel();
+            ArrayList<Listing> searchListings = responseModel.getListings();
+            setUpPanel(searchListings);
```

Comment:
I'm not sure about the architecture for this, but will the response model extend the controller? either way I'm going to assume this works